src/state/transactions/hooks.tsx

```diff
@@ -1,3 +1,4 @@
+import { BigNumber } from '@ethersproject/bignumber'
```

Comment:
Could you add unit tests for this file? It would help disambiguate approve/revoke and make sure they don't slip in the future.